cmd/lncli/cmd_macaroon.go

```diff
@@ -305,3 +307,90 @@ func listPermissions(ctx *cli.Context) error {
 
 	return nil
 }
+
+type macaroonContent struct {
+	Version     uint16   `json:"version"`
+	Location    string   `json:"location"`
+	RootKeyID   string   `json:"root_key_id"`
+	Permissions []string `json:"permissions"`
+	Caveats     []string `json:"caveats"`
+}
+
+var printMacaroonCommand = cli.Command{
+	Name:     "printmacaroon",
+	Category: "Macaroons",
+	Usage: "Print the contents of a macaroon in a human readable " +
+		"format.",
+	ArgsUsage: "mac_file",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "mac_file",
+			Usage: "the macaroon file to print",
+		},
+	},
+	Action: actionDecorator(printMacaroon),
+}
+
+func printMacaroon(ctx *cli.Context) error {
+	// Show command help if no arguments or flags are set.
+	if ctx.NArg() == 0 && ctx.NumFlags() == 0 {
+		return cli.ShowCommandHelp(ctx, "printmacaroon")
+	}
+	args := ctx.Args()
+
+	var macPath string
```

Comment:
Would be nice to not have to rely on a path and just being able to pass it hex-encoded as an arg.

Reply:
Good suggestion, added that!